### ISSUE TITLE
‘BatchNorm2d’ object has no attribute ‘track_running_stats’

### ISSUE THREAD

**Issue by Jihunlee326** [ShiyuLiang/odin-pytorch#3]
Hi,
I'am using pytorch version 0.4.0 and i get that error in that line

out = self.conv1(self.relu(self.bn1(x)))     _# line 34 on densenet.py_

The error message looks like that:

  File "/opt/odin-pytorch/code/densenet.py", line 34, in forward
    out = self.conv1(self.relu(self.bn1(x)))
  File "/usr/local/lib/python3.5/dist-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/torch/nn/modules/batchnorm.py", line 49, in forward
    self.training or not self.track_running_stats, self.momentum, self.eps)
  File "/usr/local/lib/python3.5/dist-packages/torch/nn/modules/module.py", line 532, in __getattr__
    type(self).__name__, name))

I found that other repositories have similar issues when upgrading from pytorch 0.3.0 to 0.4.0:
https://discuss.pytorch.org/t/batchnorm2d-object-has-no-attribute-track-running-stats/17525/6

Should I use pytorch 0.3.x?

**Comment by heroinlin** [comment]
Your model train by pytorch0.3.x, but run in pytorch > 0.4.0.
Change the parameter of BatchNorm2d by yourself.
For example, define the function

```
def recursion_change_bn(module):
    if isinstance(module, torch.nn.BatchNorm2d):
        module.track_running_stats = 1
    else:
        for i, (name, module1) in enumerate(module._modules.items()):
            module1 = recursion_change_bn(module1)
    return module
```
and
use it when you load model

```
check_point = torch.load(check_point_file_path)
model = check_point['net']
for i, (name, module) in enumerate(model._modules.items()):
    module = recursion_change_bn(model)
model.eval()
```
I have ran 0.3.1 model in pytorch0.4.1 and pytorch1.0.0.

**Comment by shreyas269** [comment]
`model = check_point['net']` throws a 'not subscriptable' error. 

Just replacing it by `model = check_point` loads it correctly.

**Comment by heroinlin** [comment]
yes! `model = check_point['net']` just because my code is:
```
state = {
                'net': model.module.state_dict() if use_cuda else model.state_dict(),
                'optimizer': optimizer.state_dict()
            }

save_path = 'final.pth'

torch.save(state, save_path)
```

**Comment by gauharbains** [comment]
@heroinlin Shouldn't it be `module = recursion_change_bn(module)` instead of `module = recursion_change_bn(model)` when trying to load the model ?

**Comment by heroinlin** [comment]
@gauharbains  you are right, I make a mistake.
